### ISSUE TITLE
no-variable-construct-id should not flag functions outside Construct classes

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#432]
### Describe the bug

The `no-variable-construct-id` rule incorrectly flags `new` expressions in standalone functions that are not inside a Construct class. In utility functions, using a variable for the construct ID is expected because the function is designed to be called with different IDs.

`shouldSkipIdValidation` already skips arrow functions, loops, and non-constructor methods, but does not skip regular function declarations/expressions outside Construct classes.

```typescript
// Should NOT be flagged — this is a utility function, variable ID is expected
export function createBucket(scope: Construct, id: string) {
  new Bucket(scope, id);
}

// Should be flagged — inside a Construct class constructor
export class MyConstruct extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);
    new Bucket(this, id); // ← variable ID should be flagged here
  }
}
```

### Expected Behavior

The rule should not flag `new` expressions in functions outside Construct classes.

### Current Behavior

The rule flags all `new` expressions with variable construct IDs regardless of context.

### eslint-plugin-awscdk Version

4.3.2

### Additional Context

#427 (same class of bug in `require-passing-this`, fixed in #431
